pilot/pkg/networking/core/cluster_waypoint.go

```diff
@@ -120,38 +133,131 @@ func (cb *ClusterBuilder) buildWaypointInboundVIPCluster(proxy *model.Proxy, svc
 	svcMetaList := im.Fields["services"].GetListValue()
 	svcMetaList.Values = append(svcMetaList.Values, buildServiceMetadata(svc))
 
+	// Apply DestinationRule configuration for the service
+	connectionPool, outlierDetection, loadBalancer, tls, proxyProtocol := selectTrafficPolicyComponents(policy)
+	// Add applicable metadata to the cluster to identify which config is applied for tooling
+	if policy != nil {
+		util.AddConfigInfoMetadata(localCluster.cluster.Metadata, drConfig.Meta)
+	}
+	if subset != "" {
+		util.AddSubsetToMetadata(localCluster.cluster.Metadata, subset)
+	}
+	if connectionPool == nil {
+		connectionPool = &networking.ConnectionPoolSettings{}
+	}
+
+	// For these policies, we have the standard logic apply
+	cb.applyConnectionPool(mesh, localCluster, connectionPool)
+	cb.applyH2Upgrade(localCluster, &port, mesh, connectionPool)
+	applyOutlierDetection(localCluster.cluster, outlierDetection)
+	applyLoadBalancer(localCluster.cluster, loadBalancer, &port, cb.locality, cb.proxyLabels, mesh)
```

Comment:
It is hard to say whether it is right, unless compare line to line with the sdeicar cluster bui;der

Iam wondering if it is possible to abstract fundamental helper to share with sidecar

Reply:
We maybe could but its tricky because the existing code already has a lot of `if inbound {}... if snidnat {...}`, it makes it fairly complex. Additionally, we have different constraints for PROXY and TLS. Overall I felt it was cleaner to split like this, but use the common functions (applyOutlierDetection etc)